jujugui/static/gui/src/app/components/lightbox/lightbox.js

```diff
@@ -1,35 +1,155 @@
-/* Copyright (C) 2015 Canonical Ltd. */
-
+/* Copyright (C) 2017 Canonical Ltd. */
 'use strict';
 
 const PropTypes = require('prop-types');
 const React = require('react');
+const classNames = require('classnames');
 
 const SvgIcon = require('../svg-icon/svg-icon');
 
 /**
   Renders a modal like overlay with a darkened background.
   A caption is displayed in white below the content if present.
 */
-class Lightbox extends React.PureComponent {
+class Lightbox extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      activeSlide: 0,
+      lastSlide: null
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.children.map) {
+      this.setState({
+        lastSlide: this.props.children.length - 1
+      });
+    }
+  }
+
+  _goToSlide(index) {
+    if (index < 0) {
+      index = 0;
+    }
+    if (index > this.state.lastSlide) {
+      index = this.state.lastSlide;
+    }
+    this.setState({
+      activeSlide: index
+    });
+  }
+
+  _nextSlide() {
+    const nextSlide = this.state.activeSlide + 1;
+    if (nextSlide <= this.state.lastSlide) {
+      this._goToSlide(nextSlide);
+    }
+  }
+
+  _previousSlide() {
+    const nextSlide = this.state.activeSlide - 1;
+    if (nextSlide >= 0) {
+      this._goToSlide(nextSlide);
+    }
+  }
+
+  _generateNavigation() {
+    if (!this.state.lastSlide) {
+      return;
+    }
+
+    const bullets = [];
+
+    for(let i = 0, ii = this.state.lastSlide; i <= ii; i += 1) {
+      const classes = classNames(
+        'lightbox__navigation-bullet',
+        {
+          'is-active': this.state.activeSlide === i
+        }
+      );
+
+      bullets.push(
+        <li
+          className={classes}
+          key={i}
+          onClick={this._goToSlide.bind(this, i)}>&bull;</li>);
+    }
+
+    return (
+      <div className="lightbox__navigation">
+        <button
+          disabled={this.state.activeSlide === 0}
+          onClick={this._previousSlide.bind(this)}
```

Comment:
instead of having a prev/next slide methods you could instead call the `_goToSlide` method directly with a delta since you're doing the same work multiple times with this approach...

```js
 _goToSlide(delta, index) { ... }

onClick={this._goToSlide.bind(this, -1)}
```

Reply:
Done